### PR TITLE
Hazelcast implementation

### PR DESCRIPTION
**Pull request by hennihaus**

### CODE REVIEW

File: src/main/java/de/blackforestsolutions/dravelopsdatamodel/Leg.java [hunk 1]
```diff
@@ -1,49 +1,55 @@
...
+    private TravelProvider travelProvider;
 
-    private final String vehicleNumber;
+    private String vehicleNumber;
 
-    private final String vehicleName;
+    private String vehicleName;
 
-    private final LinkedList<TravelPoint> intermediateStops;
+    private LinkedList<TravelPoint> intermediateStops;
```
Comment (mburkertt) [inline src/main/java/de/blackforestsolutions/dravelopsdatamodel/Leg.java:50]: Maybe we need this to implement Portable, but we arent immutable anymore in this case. How do we enforce to not use setter for this method?

File: src/main/java/de/blackforestsolutions/dravelopsdatamodel/Price.java [hunk 2]
```diff
@@ -1,29 +1,82 @@
...
+    private static final long serialVersionUID = -3355934959094849304L;
 
-    private final PriceType priceType;
+    private PriceType priceType;
 
-    private final Currency currencyCode;
+    private Currency currencyCode;
 
-    private final Long smallestCurrencyValue;
+    private Long smallestCurrencyValue;
```
Comment (mburkertt) [inline src/main/java/de/blackforestsolutions/dravelopsdatamodel/Price.java:31]: Maybe we need this to implement Portable, but we arent immutable anymore in this case. How do we enforce to not use setter for this method?

File: src/main/java/de/blackforestsolutions/dravelopsdatamodel/TravelPoint.java [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
+    private Point point;
 
-    private final ZonedDateTime arrivalTime;
+    private ZonedDateTime arrivalTime;
 
-    private final ZonedDateTime departureTime;
+    private ZonedDateTime departureTime;
 
-    private final String platform;
+    private String platform;
```
Comment (mburkertt) [inline src/main/java/de/blackforestsolutions/dravelopsdatamodel/TravelPoint.java:35]: Maybe we need this to implement Portable, but we arent immutable anymore in this case. How do we enforce to not use setter for this method?

File: src/main/java/de/blackforestsolutions/dravelopsdatamodel/TravelProvider.java [hunk 4]
```diff
@@ -1,28 +1,67 @@
...
-public final class TravelProvider {
+public final class TravelProvider implements Serializable, Portable {
+
+    private static final long serialVersionUID = -8728155372687060904L;
 
-    private final String name;
+    private String name;
 
-    private final URL url;
+    private URL url;
```
Comment (mburkertt) [inline src/main/java/de/blackforestsolutions/dravelopsdatamodel/TravelProvider.java:29]: Maybe we need this to implement Portable, but we arent immutable anymore in this case. How do we enforce to not use setter for this method?

File: src/test/java/de/blackforestsolutions/dravelopsdatamodel/JourneyTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+import static de.blackforestsolutions.dravelopsdatamodel.objectmothers.JourneyObjectMother.getJourneyBuilderWithNoEmptyFields;
+import static de.blackforestsolutions.dravelopsdatamodel.objectmothers.JourneyObjectMother.getJourneyWithEmptyFields;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class JourneyTest {
+
+    @Test
+    void test_build_journey_generates_conform_sha1Id() throws IOException {
+        Pattern sha1IdPattern = Pattern.compile("[0-9a-f]{40}");
```
Comment (mburkertt) [inline src/test/java/de/blackforestsolutions/dravelopsdatamodel/JourneyTest.java:18]: We use this regex here which is ok but we should comment it to be sure or shouldnt we ? 
For me it reads out the following way:

All small written letters and numbers are allowed and it was to be 40 sings long.

Is my reading correct ?

**Review by mburkertt — CHANGES_REQUESTED** [review]
There are some questions open.

File: src/main/java/de/blackforestsolutions/dravelopsdatamodel/Leg.java (hunk 1 above)
Comment (hennihaus) [reply src/main/java/de/blackforestsolutions/dravelopsdatamodel/Leg.java:50]: Wir sollten meiner Meinung nach Portable wieder entfernen. Ich habe noch keine Moeglichkeit gefunden Portable in der Server-Client-Arhcitektur zu konfigurieren.

Comment (hennihaus) [reply src/main/java/de/blackforestsolutions/dravelopsdatamodel/Leg.java]: Done

File: src/main/java/de/blackforestsolutions/dravelopsdatamodel/Price.java (hunk 2 above)
Comment (hennihaus) [reply src/main/java/de/blackforestsolutions/dravelopsdatamodel/Price.java]: Done

File: src/main/java/de/blackforestsolutions/dravelopsdatamodel/TravelProvider.java (hunk 4 above)
Comment (hennihaus) [reply src/main/java/de/blackforestsolutions/dravelopsdatamodel/TravelProvider.java]: Done

File: src/test/java/de/blackforestsolutions/dravelopsdatamodel/JourneyTest.java (hunk 5 above)
Comment (blackforestsolutions) [reply src/test/java/de/blackforestsolutions/dravelopsdatamodel/JourneyTest.java:23]: Done

File: src/main/java/de/blackforestsolutions/dravelopsdatamodel/TravelPoint.java (hunk 3 above)
Comment (blackforestsolutions) [reply src/main/java/de/blackforestsolutions/dravelopsdatamodel/TravelPoint.java:34]: Done

**Pull request merged**